roofit/roofitcore/inc/RooMinimizer.h

```diff
@@ -20,147 +20,168 @@
-  void setEvalErrorWall(bool flag) ;
-  void setRecoverFromNaNStrength(double strength) ;
-  void setOffsetting(bool flag) ;
-  void setMaxIterations(int n) ;
-  void setMaxFunctionCalls(int n) ;
-
-  int migrad() ;
-  int hesse() ;
-  int minos() ;
-  int minos(const RooArgSet& minosParamList) ;
-  int seek() ;
-  int simplex() ;
-  int improve() ;
-
-  int minimize(const char* type, const char* alg=nullptr) ;
-
-  RooFitResult* save(const char* name=nullptr, const char* title=nullptr) ;
-  RooPlot* contour(RooRealVar& var1, RooRealVar& var2,
-         double n1=1.0, double n2=2.0, double n3=0.0,
-         double n4=0.0, double n5=0.0, double n6=0.0, unsigned int npoints = 50) ;
-
-  int setPrintLevel(int newLevel) ;
-  void setPrintEvalErrors(int numEvalErrors) ;
-  void setVerbose(bool flag=true) ;
-  void setProfile(bool flag=true) { _profile = flag ; }
-  bool setLogFile(const char* logf=nullptr) ;
-
-  /// Enable or disable the logging of function evaluations to a RooDataSet.
-  /// \see RooMinimizer::getLogDataSet().
-  /// param[in] flag Boolean flag to disable or enable the functionality.
-  void setLoggingToDataSet(bool flag=true) { _loggingToDataSet = flag ; }
-
-  /// If logging of function evaluations to a RooDataSet is enabled, returns a
-  /// pointer to a dataset with one row per evaluation of the RooAbsReal passed
-  /// to the minimizer. As columns, there are all floating parameters and the
-  /// values they had for that evaluation.
-  /// \see RooMinimizer::setLoggingToDataSet(bool).
-  RooDataSet * getLogDataSet() const { return _logDataSet.get(); }
-
-  int getPrintLevel() const { return _printLevel; }
-
-  void setMinimizerType(std::string const& type) ;
-  std::string const& minimizerType() const { return _minimizerType; }
-
-  static void cleanup() ;
-  static RooFitResult* lastMinuitFit() ;
-  static RooFitResult* lastMinuitFit(const RooArgList& varList) ;
-
-  void saveStatus(const char* label, int status) { _statusHistory.push_back(std::pair<std::string,int>(label,status)) ; }
-
-  int evalCounter() const ;
-  void zeroEvalCount() ;
-
-  ROOT::Fit::Fitter* fitter() ;
-  const ROOT::Fit::Fitter* fitter() const ;
-
-  ROOT::Math::IMultiGenFunction* getMultiGenFcn() const;
-
-  int getNPar() const ;
-
-  void applyCovarianceMatrix(TMatrixDSym const& V) ;
+   enum class FcnMode { classic, gradient, generic_wrapper };
+
+   /// Config argument to RooMinimizer ctor
+   struct Config {
+      double recoverFromNaN = 10.; // RooAbsMinimizerFcn config
+      int optConst = 0;            // RooAbsMinimizerFcn config
+      int printEvalErrors = 10;    // RooAbsMinimizerFcn config
+      int doEEWall = 1;            // RooAbsMinimizerFcn config
+      int offsetting = -1;         // RooAbsMinimizerFcn config
+      const char *logf = nullptr;  // RooAbsMinimizerFcn config
+      int nWorkers =
+         RooFit::MultiProcess::Config::getDefaultNWorkers(); // RooAbsMinimizerFcn config that can only be set in ctor
```

Comment:
This has to be refactored, because right now it necessitates the inclusion of a private header.

I would do it like this:
* call a `private` static function of `RooMinimizer::Config`:
  `int nWorkers = getDefaultWorkers();`
* define this function in `RooMinimizer.cxx`, again with the proper include guards:
   ```
    int RooMinimizer::Config::getDefaultWorkers() {
    #ifdef R__HAS_ROOFIT_MULTIPROCESS
       return RooFit::MultiProcess::Config::getDefaultNWorkers();
    #else
       return 0;
    #endif
    }
    ```